### PR TITLE
Add tests for the cljs.spec.test.alpha/instrument WARNING #308

### PR DESCRIPTION
**Pull request by arichiardi**

### CODE REVIEW

**Comment by arichiardi** [comment]
Will fix the indentation.

File: src/test/lumo/lumo/build_api_tests.cljs [hunk 1]
```diff
@@ -376,12 +376,39 @@
...
+        warning-handlers [(fn [warning-type env extra]
+                            (let [w (warning-type ana/*cljs-warnings*)
+                                  err (ana/error-message warning-type extra)]
+                              (println "WARNING:" (ana/message env err))
+                              (is (nil? warning-type) "when compiling twice, it should not emit a WARNING for cljs.spec.test.alpha/instrument the second time")))]]
+    (testing "correctly cljs.js/ns-side-effects on read analysis cache"
+      (test/delete-out-files out)
+      (build/build
+       (path/join root "instrument")
+       {:main 'test-check.instrument.core
```
Comment (anmonteiro) [inline src/test/lumo/lumo/build_api_tests.cljs]: `instrument.core`

File: src/test/lumo/lumo/build_api_tests.cljs [hunk 2]
```diff
@@ -376,12 +376,39 @@
...
       (test/delete-out-files out)
       (is (nil? (build/build
-                  (path/join root "test_check")
-                  {:main 'hello
-                   :optimizations :none
-                   :output-to out}
-                  (env/default-compiler-env))) "It should successfully compile with :optimizations :none")
-        (test/delete-out-files out))))
+                 (path/join root "test_check")
+                 {:main 'test-check.summary.core
```
Comment (anmonteiro) [inline src/test/lumo/lumo/build_api_tests.cljs]: why did you change this test?

File: src/test/lumo/lumo/build_api_tests.cljs [hunk 3]
```diff
@@ -376,12 +376,39 @@
...
+      (build/build
+       (path/join root "instrument")
+       {:main 'test-check.instrument.core
+        :optimizations :none
+        :warning-handlers warning-handlers
+        :output-to out}
+       (env/default-compiler-env))
+      (build/build
+       (path/join root "instrument")
+       {:main 'test-check.instrument.core
```
Comment (anmonteiro) [inline src/test/lumo/lumo/build_api_tests.cljs]: `instrument.core`

File: src/test/lumo/lumo/build_api_tests.cljs (hunk 2 above)
Comment (arichiardi) [reply src/test/lumo/lumo/build_api_tests.cljs]: Sorry it is not changed I will revert...probably some copy and paste in the wrong place

**Comment by arichiardi** [comment]
Awesome!
